Isodose/qSlicerIsodoseModuleWidget.cxx

```diff
@@ -430,6 +440,56 @@ void qSlicerIsodoseModuleWidget::setDoseVolumeNode(vtkMRMLNode* node)
   if (paramNode->GetDoseVolumeNode() && vtkSlicerRtCommon::IsDoseVolumeNode(paramNode->GetDoseVolumeNode()))
   {
     d->label_NotDoseVolumeWarning->setText("");
+
+    vtkMRMLSubjectHierarchyNode* shNode = vtkMRMLSubjectHierarchyNode::GetSubjectHierarchyNode(this->mrmlScene());
+    if (!shNode)
+    {
+      qCritical() << Q_FUNC_INFO << ": Failed to access subject hierarchy node";
+      return;
+    }
+
+    std::string doseUnitName("");
+    vtkIdType doseShItemID = shNode->GetItemByDataNode(paramNode->GetDoseVolumeNode());
+    if (doseShItemID != vtkMRMLSubjectHierarchyNode::INVALID_ITEM_ID)
+    {
+      doseUnitName = shNode->GetAttributeFromItemAncestor(
+        doseShItemID, vtkSlicerRtCommon::DICOMRTIMPORT_DOSE_UNIT_NAME_ATTRIBUTE_NAME, vtkMRMLSubjectHierarchyConstants::GetDICOMLevelStudy());
+    }
+
+    double valueRange[2];
+    vtkImageData* image = paramNode->GetDoseVolumeNode()->GetImageData();
+    image->GetScalarRange(valueRange);
+    if (!doseUnitName.compare("RELATIVE"))
+    {
+      paramNode->SetDoseUnits(vtkMRMLIsodoseNode::RELATIVE);

```

Comment:
This is how you should namespace an enum (the class name instead of the enum type). Please fix the wrong ones.